openscad.pro

```diff
@@ -218,7 +218,7 @@ RESOURCES = openscad.qrc
 # Qt5 removed access to the QMAKE_UIC variable, the following
 # way works for both Qt4 and Qt5
 load(uic)
-uic.commands += -tr _
+uic.commands += -tr tr
```

Comment:
I just tried commenting out line 220 and 221.

This is an extract from ui_MainWindow.h from a
"normal current build":
```
    void retranslateUi(QMainWindow *MainWindow)
    {
        fileActionNew->setText(_("&New", nullptr));
#ifndef QT_NO_SHORTCUT
        fileActionNew->setShortcut(_("Ctrl+N", nullptr));
#endif // QT_NO_SHORTCUT
```
this is with -tr tr:
```
    void retranslateUi(QMainWindow *MainWindow)
    {
        fileActionNew->setText(tr("&New", nullptr));
#ifndef QT_NO_SHORTCUT
        fileActionNew->setShortcut(tr("Ctrl+N", nullptr));
#endif // QT_NO_SHORTCUT
```
this is without line 220 and 221 in openscad.pro:
```
        fileActionNew->setText(QApplication::translate("MainWindow", "&New", nullptr));
#ifndef QT_NO_SHORTCUT
        fileActionNew->setShortcut(QApplication::translate("MainWindow", "Ctrl+N", nullptr));
#endif // QT_NO_SHORTCUT
```
Kind of explained in:
http://doc.qt.io/qt-5/qcoreapplication.html#translate

Also note that even if QT would use tr(), it would not use our tr() in `src/qtgettext.h`, but essentiall this->tr():

http://doc.qt.io/qt-5/i18n-source-translation.html

> ```
> void MainWindow::createMenus()
> {
>     fileMenu = menuBar()->addMenu(tr("&File"));
>     ...
> ```
> Here, the translation context is MainWindow because it is the MainWindow::tr() function that is invoked. 

Okay - now I understand the issue:

http://doc.qt.io/qt-5/qobject.html#tr

This is ambiguous and I think our implementation is only called, because the parameter count (two) matches, where as QTs tr has three parameters, two of them beeing optional.

Reply:
now it is q_() as in **_**() returning a **Q**String

Reply:
@t-paul us q_() fine with you?

Reply:
Sure, sounds good. The main thing is that it's different from the defaults so we would catch unintended mixups.